arwen/elrondapi/elrondei.go

```diff
@@ -914,6 +934,99 @@ func v1_4_getESDTTokenData(
 	return int32(len(esdtData.Value.Bytes()))
 }
 
+//export v1_4_getESDTLocalRoles
+func v1_4_getESDTLocalRoles(context unsafe.Pointer, tokenIdHandle int32) int64 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	storage := arwen.GetStorageContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	tokenID, err := managedType.GetBytes(tokenIdHandle)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return -1
+	}
+	key := []byte("ELRONDroleesdt" + string(tokenID))
+
+	data_buffer, usedCache := storage.GetStorage(key)
+	storage.UseGasForStorageLoad(
+		storageLoadName,
+		metering.GasSchedule().ElrondAPICost.StorageLoad,
+		len(data_buffer), usedCache)
+
+	result := int64(0)
```

Comment:
make a separate function out of lines 956-974 and you will be able to easily unit test